### PR TITLE
feat: simulation types + cleanup

### PR DESCRIPTION
**Pull request by FrankieIsLost**
Add type implementations for `mev_simBundle` endpoint. Also some general cleanup of CI & README

### CODE REVIEW

File: crates/mev-share-rpc-api/src/mev.rs [hunk 1]
```diff
@@ -16,5 +16,5 @@ pub trait MevApi {
     /// Similar to `mev_sendBundle` but instead of submitting a bundle to the relay, it returns a
     /// simulation result. Only fully matched bundles can be simulated.
     #[method(name = "simBundle")]
-    async fn sim_bundle(&self, request: SendBundleRequest) -> RpcResult<()>;
+    async fn sim_bundle(&self, bundle: SendBundleRequest, sim_options: SimBundleOptions) -> RpcResult<SimBundleResponse>;
```
Comment (mattsse) [inline crates/mev-share-rpc-api/src/mev.rs:19]: are these documented anywhere?

couldn't find it here

https://docs.flashbots.net/flashbots-auction/searchers/advanced/rpc-endpoint#mev_simbundle

File: crates/mev-share-rpc-api/src/types.rs [hunk 2]
```diff
@@ -267,6 +267,71 @@ pub enum ProtocolVersion {
     V0_1,
 }
 
+/// Optional fields to override simulation state.
+#[derive(Deserialize, Debug, Serialize, Clone, Default, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct SimBundleOptions {
```
Comment (mattsse) [inline crates/mev-share-rpc-api/src/types.rs:273]: perhaps we should call these Overrides?

File: crates/mev-share-rpc-api/src/mev.rs (hunk 1 above)
Comment (FrankieIsLost) [reply crates/mev-share-rpc-api/src/mev.rs:19]: I don't think these are documented, I pulled them from the reference ts client implementation https://github.com/flashbots/mev-share-client-ts/blob/main/src/client.ts#L195

Comment (FrankieIsLost) [reply crates/mev-share-rpc-api/src/mev.rs:19]: Added an issue in the docs to get them added https://github.com/flashbots/flashbots-docs/issues/403

File: Cargo.toml [hunk 3]
```diff
@@ -17,8 +17,8 @@ edition = "2021"
 rust-version = "1.70"
 authors = ["Matthias Seitz <matthias.seitz@outlook.de>"]
```
Comment (mattsse) [inline Cargo.toml:18]: can also update this to mev-share-rs devs or something

**Review by mattsse — APPROVED** [review]
lgtm

File: crates/mev-share-rpc-api/src/mev.rs (hunk 1 above)
Comment (odysseus0) [reply crates/mev-share-rpc-api/src/mev.rs]: Adding it now

**Pull request merged**